irohad/network/impl/block_loader_impl.cpp

```diff
@@ -105,25 +113,32 @@ nonstd::optional<Block> BlockLoaderImpl::retrieveBlock(
     return nonstd::nullopt;
   }
 
-  auto &&result = factory_.deserialize(block);
-  if (not crypto_provider_->verify(result)) {
+  auto result = std::make_shared<shared_model::proto::Block>(block);
+  std::unique_ptr<iroha::model::Block> old_block(result->makeOldModel());
+  if (not crypto_provider_->verify(*old_block)) {
     log_->error(kInvalidBlockSignatures);
     return nonstd::nullopt;
   }
 
-  return result;
+  return static_cast<Commit>(result);
```

Comment:
Also, looks like a bug.

Reply:
That one is required, it has 2 casts. First proto->iface, then iface->optional